libvips/foreign/magick.c

```diff
@@ -83,8 +83,9 @@ magick_sniff(const unsigned char *bytes, size_t length)
 	if (length >= 4 &&
 		bytes[0] == 0 &&
 		bytes[1] == 0 &&
-		bytes[2] == 1 &&
-		bytes[3] == 0)
+		(bytes[2] == 1 || bytes[2] == 2) &&
+		bytes[3] == 0 &&
+		!(bytes[4] == 0 && bytes[5] == 0))
```

Comment:
nit: I would write this as:
```suggestion
		(bytes[4] != 0 || bytes[5] != 0)
```